include/zephyr/net/socket_can.h

```diff
@@ -51,6 +52,133 @@ struct sockaddr_can {
 	int         can_ifindex;
 };
 
+/**
+ * @name Linux SocketCAN compatibility
+ *
+ * The following structures and functions provide compatibility with the CAN
+ * frame and CAN filter formats used by Linux SocketCAN.
+ *
+ * @{
+ */
+
+/**
+ * CAN Identifier structure for Linux SocketCAN compatibility.
+ *
+ * The fields in this type are:
+ *
+ * @code{.text}
+ *
+ * +------+--------------------------------------------------------------+
+ * | Bits | Description                                                  |
+ * +======+==============================================================+
+ * | 0-28 | CAN identifier (11/29 bit)                                   |
+ * +------+--------------------------------------------------------------+
+ * |  29  | Error message frame flag (0 = data frame, 1 = error message) |
+ * +------+--------------------------------------------------------------+
+ * |  30  | Remote transmission request flag (1 = RTR frame)             |
+ * +------+--------------------------------------------------------------+
+ * |  31  | Frame format flag (0 = standard 11 bit, 1 = extended 29 bit) |
+ * +------+--------------------------------------------------------------+
+ *
+ * @endcode
+ */
+typedef uint32_t socketcan_id_t;
+
+/**
+ * @brief CAN frame for Linux SocketCAN compatibility.
+ */
+struct socketcan_frame {
```

Comment:
I wonder if it would make sense to put this struct definition into a separate file, which can be included here and in the Linux SockedCAN "driver" I'm working on. That way we could use it as the exchange format without automatically including `zephyr/drivers/can.h`, which is required here for the conversion functions, but which has conflicting struct definitions when also the native linux can headers are included.

Reply:
Separating the two is a good idea. Either a separate header file for the `struct socketcan_*` stuff or we could move the `socketcan_*` utility functions (which require `can.h`) to a separate include file.

Which would you prefer? And what naming?

Reply:
Moving the utility functions into a file `socketcan_utils.h` would be somewhat consistent with the `can_utils.h` in the driver (even if the latter one is a local include file only). Maybe also rename `socket_can.h` to `socketcan.h` for more consistency while we are at it?

Reply:
`socket_can.h` is named like that to be consistent with the other `socket_*.h` files, but perhaps we can change it.

Reply:
Don't have a strong opinion here, both fine for me.

Reply:
I'll give it some thought and come up with a suggestion. Thanks.

Reply:
@martinjaeger I have renamed the header as suggested and split the utility functions into their own header.